Add context around AWS join methods (#43380)
gravitational/teleport: docs/pages/agents/join-services-to-your-cluster/aws-ec2.mdx

description: Use the EC2 join method to add services to your Teleport cluster on AWS
---

This guide explains how to use the **EC2 join method** to configure Teleport

> processes to join your Teleport cluster without sharing any secrets when they a
> re running in AWS.

Reviewer: ```suggestion
processes to join your Teleport cluster without sharing any secrets when they
are running in AWS.
```

> The EC2 join method is only available in self-hosted Teleport deployments. There

Reviewer 2: This definitely helps elucidate the difference between the IAM and EC2 join methods, but it doesn't really provide any suggestion as to when a specific join method may be better. I'm of the slight opinion that we ought to push people towards the IAM join method as it seems generally better supported and the EC2 join method has a number of papercuts.

Author: 👍 What makes you say that the IAM method is better supported? Are there any papercuts with the EC2 method that we should mention? I'm also generally not sure why someone would opt for the EC2 method over the IAM method—are there any situations where attaching an IAM identity to an Agent is less convenient than attaching one to the Auth Service while Agents are running on EC2?

Reviewer 2: IAM method also integrates nicely with IRSA on EKS, and is also supported on Teleport Cloud or Self-hosted. IIRC the EC2 method came first, and eventually the IAM method was introduced because of the limitations of the EC2 method.

Author: Thanks!

are two other AWS join methods available depending on your use case:

1. The **IAM join method** is available to any Teleport process running anywhere
   with access to IAM credentials, such as an EC2 instance with an attached IAM

>    role (see [documentation](aws-iam.mdx). No specific permissions or IAM policy

Reviewer: link looks wrong, maybe this?
```suggestion
   role (see [documentation](aws-iam.mdx)). No specific permissions or IAM policy
```
